packages/angular_devkit/build_angular/src/dev-server/index.ts

```diff
@@ -305,17 +305,19 @@ export function serveWebpackBrowser(
     switchMap(({ browserOptions, webpackConfig, locale }) => {
       if (browserOptions.index) {
         const { scripts = [], styles = [], baseHref } = browserOptions;
-        const entrypoints = generateEntryPoints({ scripts, styles });
+        const entrypoints = generateEntryPoints({
+          scripts,
+          styles,
+          isHMREnabled: webpackConfig.devServer?.hot,
```

Comment:
Can you add a comment here explain why this HMR check is currently needed?
Hopefully we can remove it in the future.

Reply:
Sure